src/main/java/teammates/logic/core/FeedbackResponsesLogic.java

```diff
@@ -55,30 +58,52 @@ public void createFeedbackResponse(FeedbackResponseAttributes fra)
             }
         }
     }
-
+    
     public FeedbackResponseAttributes getFeedbackResponse(
             String feedbackResponseId) {
-        return frDb.getFeedbackResponse(feedbackResponseId);
+        FeedbackResponseAttributes response = frDb.getFeedbackResponse(feedbackResponseId);
+        if (response == null) {
+            return null;
+        }
```

Comment:
Since other places will return a list of responses, every response will be checked whether it is null in isResponseValid method.